service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/CreateAndDeleteAzureControlledResourceFlightTest.java

```diff
@@ -61,23 +55,61 @@ public class CreateAndDeleteAzureControlledResourceFlightTest extends BaseAzureT
   @Autowired private ControlledResourceService controlledResourceService;
   @Autowired private WsmResourceService wsmResourceService;
 
-  @Test
-  public void createAzureIpControlledResource() throws InterruptedException {
-    UUID workspaceUuid = azureTestUtils.createWorkspace(workspaceService);
-    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
-
-    // Cloud context needs to be created first
-    FlightState createAzureContextFlightState =
-        StairwayTestUtils.blockUntilFlightCompletes(
+  private void createCloudContext(UUID workspaceUuid, AuthenticatedUserRequest userRequest)
```

Comment:
Thanks for abstracting all of this out! 🎉

Reply:
I felt the need to reduce my guilt for copy/pasting so much in this PR. :)